tests/functions/folding/test_fold_as_wei_value.py

```diff
@@ -0,0 +1,60 @@
+from hypothesis import (
+    given,
+    settings,
+    strategies as st,
+)
+import pytest
+
+from vyper import (
+    ast as vy_ast,
+    functions as vy_fn,
+)
+
+denoms = [x for k in vy_fn.AsWeiValue.wei_denoms.keys() for x in k]
+
+
+st_decimals = st.decimals(
+    min_value=0,
+    max_value=2 ** 32,
```

Comment:
Probably don't need to do anything more than like 10 or 100. We want some < 1.0 values

Reply:
Note: Would fail because of https://github.com/vyperlang/vyper/issues/1637